barretenberg/cpp/src/barretenberg/stdlib/hash/poseidon2/sponge/sponge.hpp

```diff
@@ -135,7 +135,7 @@ template <size_t rate, size_t capacity, size_t t, typename Permutation, typename
      * @return std::array<field_t, out_len>
      */
     template <size_t out_len, bool is_variable_length>
-    static std::array<field_t, out_len> hash_internal(Builder& builder, std::span<field_t> input)
+    static std::array<field_t, out_len> hash_internal(Builder& builder, std::span<const field_t> input)
```

Comment:
allows for conversion from const std::vector to std::span. Otherwise, it throws an error.